### PR TITLE
Added feature to search Google for image

### PR DESCRIPTION
**Pull request by innermous**
This feature is really lacking one :)

### CODE REVIEW

**Comment by ke7ofi** [comment]
Is this something that should be built into the browser? It seems like something better suited to a addon or JS thing.

**Comment by innermous** [comment]
I asked maintainer before PR, he said this sounds good. 
Can't see this as addon tho, this functionality is pretty common.

**Comment by cranes-bill** [comment]
What does mean „_pretty common_“!? Firefox doesn't have this function. I can't say about other browsers for sure, but i think more of them also don't have it. Even there is a Chrome [add-on](https://chrome.google.com/webstore/detail/search-by-image-by-google/dajedkncpodkggklbegccjpmnglmnflm), so probably even Chrome don't have this as a commonly  function. Furthermore, some people might prefer different search engine.....

**Comment by ke7ofi** [comment]
That's the other thing - keeping things engine-agnostic is good.

File: src/lib/webengine/webview.cpp [hunk 1]
```diff
@@ -728,6 +728,12 @@ void WebView::createImageContextMenu(QMenu* menu, const WebHitTestResult &hitTes
     connect(act, SIGNAL(ctrlTriggered()), this, SLOT(userDefinedOpenUrlInNewTab()));
     menu->addAction(act);
     menu->addAction(QIcon::fromTheme("edit-copy"), tr("Copy image ad&dress"), this, SLOT(copyLinkToClipboard()))->setData(hitTest.imageUrl());
+    QUrl imageSearchUrl("https://www.google.com/searchbyimage?site=search&image_url=" + act->data().toString());
```
Comment (nowrep) [inline src/lib/webengine/webview.cpp]: Use `hitTest.imageUrl()`. Also, this may need to be percent encoded.

**Comment by innermous** [comment]
All right, I think you are right, we need sort of browser add-on here.

**Comment by nowrep** [comment]
I didn't say to close it.
I think it is useful feature, and even though it would be great if it were search provider agnostic, the fact is, that only google supports it. So there is not much else to do (same as the actions for translate).

**Comment by cranes-bill** [comment]
@nowrep says: ... the fact is, that only google supports it.....
I'm not a quite agree with that.

**Comment by ke7ofi** [comment]
@nowrep @cranes-bill There's TinEye, for example.

**Comment by innermous** [comment]
@nowrep bing, yandex, and so on...
I thought, that it would be better if we make this feature plug-in, giving user a choice, whether he wanna use it at all, if so - what search engine is preferred... There's a little room for user options, but if u insist, I'll just edit PR and that will do.

**Pull request closed without merging**